Mathlib/Algebra/Module/ZLattice/Covolume.lean

```diff
@@ -130,9 +134,52 @@ theorem covolume_eq_det_inv {ι : Type*} [Fintype ι] (L : Submodule ℤ (ι →
     IsUnit.unit_spec, ← Basis.det_basis, LinearEquiv.coe_det]
   rfl
 
-theorem volume_image_eq_volume_div_covolume {ι : Type*} [Fintype ι]
-    (L : Submodule ℤ (ι → ℝ)) [DiscreteTopology L] [IsZLattice ℝ L] (b : Basis ι ℤ L)
-    {s : Set (ι → ℝ)} :
+/--
+Let `L₁` be a sub-`ℤ`-lattice of `L₂`. Then the index of `L₁` inside `L₂` is equal to
+`covolume L₁ / covolume L₂`.
+-/
+theorem covolume_div_covolume_eq_relindex {ι : Type*} [Fintype ι] [DecidableEq ι]
```

Comment:
Can you instead use `classical` in the proof?
```suggestion
theorem covolume_div_covolume_eq_relindex {ι : Type*} [Fintype ι]
```

Reply:
Good point.